transformer_sort_test.go

```diff
@@ -131,6 +132,183 @@ func TestSortWithRowKey(t *testing.T) {
 	}
 }
 
+func TestSortWithTime(t *testing.T) {
+	data, err := filtrify.ConvertToTypedData(test.UAT1TestDataFormattedWithTime, true, true, true)
+	if err != nil {
+		assert.NoError(t, err, "basic data conversion failed")
+	}
+
+	orderedColumn := "Active From"
+
+	conf := &operator.SortConfiguration{
+		OrderBy: []*operator.OrderConfiguration{
+			{
+				ColumnName: orderedColumn,
+				Ascending:  false,
+			},
+		},
+	}
+	b1, err := json.Marshal(conf)
+	if err != nil {
+		panic(err.Error())
+	}
+	step := &types.TransformationStep{
+		Operator:      types.Sort,
+		Configuration: string(b1),
+	}
+
+	sortedData, err := filtrify.Transform(data, []*types.TransformationStep{step}, nil)
+	if err != nil {
+		assert.NoError(t, err, "sort column operation failed")
+	}
+
+	var lastVal *time.Time = nil
+	firstCol := test.GetColumn(sortedData.Rows[0], orderedColumn)
+	assert.NotNil(t, firstCol, fmt.Sprintf("%s column was not found", orderedColumn))
+	if firstCol.CellValue.DataType != types.NilType {
+		lastVal = &firstCol.CellValue.TimestampValue
+	}
+	for _, r := range sortedData.Rows {
+		sortedColValue := test.GetColumn(r, orderedColumn)
+		assert.NotNil(t, sortedColValue, fmt.Sprintf("%s column was not found", orderedColumn))
+
+		if lastVal == nil && sortedColValue.CellValue.DataType != types.NilType {
+			assert.Fail(t, "descending sort failed. numbers can't appear after nil values")
+		}
+
+		if lastVal != nil {
+			isAfter := lastVal.After(sortedColValue.CellValue.TimestampValue)
+			isSame := lastVal.Equal(sortedColValue.CellValue.TimestampValue)
+			assert.True(t, isAfter || isSame, "descending order failed")
+		}
+
+		if sortedColValue.CellValue.DataType == types.NilType {
+			lastVal = nil
+		} else if lastVal == nil {
+			lastVal = &sortedColValue.CellValue.TimestampValue
+		}
+
+	}
```

Comment:
i wont be fighting you about it but i think it would be better to just check the dates in the data set appears in the correct order hard coded, now you have to parse the test code, maybe there is a bug somewhere? Could the times be time.Zero was a thought i had for example? Probably handled somewhere but not clear at first glance to me

Reply:
i think that will be nice, hardcoding some data that is not parsed. but this is also useful - this way we also know that parsing function works properly. which can be used from upper layers. will add another hardcoded data set too

Reply:
Yep! But you cand hard code time.Time values as well and diff the utc value with a certain formatter for example

Reply:
i did hardcode time.Time values. but i don't understand - diffing the utc value with a certain formatter part. can you describe it?

Reply:
Just diffing time.Time with assert.Equals often results in diffs even if the time is the same instant, and even if both are utc, depending on how the time instance was initialized, so just a way to standardise for the diff